### ISSUE TITLE
Ctrl+W doesn't delete previous word on Input

### ISSUE THREAD

**Issue by ahmetb** [AlecAivazis/survey#86]
Answer area to the survey.Input question cannot be deleted with the Ctrl+W key (which is supported by most terminal emulators, such as iTerm).

**Comment by AlecAivazis** [comment]
Thanks for showing me this shortcut, I didn't know about it at all. That being said, the testing a did on my OSX machines showed that it removes the full line, not just the last word. On my windows machine, I do not get any response. And I do not have a linux machine to test this on. Do you by chance have access to a machine with linux? If not, I can verify the right behavior on Monday using a coworkers machine.

**Comment by ahmetb** [comment]
Mac and Linux behavior should be the same. However, ctrl+W removes only a word, not the whole line. There are other shortcuts too. Maybe this library should just let the terminal emulator do its job (if possible) instead of handling the shortcuts?

**Comment by AlecAivazis** [comment]
Ah you're right - not sure what I was seeing last night. 

Also, unfortunately, we had to "own" all of the interaction with the keyboard in order to have reliable support for windows environments as well as posix ones which means we don't have an easy way to have the terminal perform the default behavior. Maybe @coryb has an idea, but my hunch is that it would be easier to add support for these shortcuts when we find them missing.

**Comment by coryb** [comment]
Yeah, I had some complaints from my users as well about missing standard control sequences, so I started to take a look.  I got as far as defining what was missing, but I have not had time to get back to implement the missing sequences.  This is what I have defined for my `survey/terminals/sequences.go`:
```
    // cursor movement
    KeyMoveRightChar = '\x06' // ctrl-f forward one char or right-arrow ^[[C
    KeyMoveRightWord = 0xe000 // alt-f foward one word
    KeyMoveRightLine = '\x05' // ctrl-e jump to end of line
    KeyMoveLeftChar  = '\x02' // ctrl-b backwards one char or left-arrow ^[[D
    KeyMoveLeftWord  = 0xe001 // alt-b backwards one word (or start of word)
    KeyMoveLeftLine  = '\x01' // ctrl-a jump to beginning of line
    KeyMoveUpLine   = '\x10' // ctrl-p previous line or up arrow ^[[A
    KeyMoveDownLine = '\x0e' // ctrl-n next line or down arrow ^[[B

    KeyCapitalizeWord = 0xe002 // alt-c capitalized from cursor to end of word
    KeyLowercaseWord  = 0xe003 // alt-l make lower case from cusror to end of word
    KeyUppercaseWord  = 0xe004 // alt-u make upper case from cusor to end of word

    KeySwapLeftChar = '\x14' // ctrl-t transpose characters
    KeySwapLeftWord = 0xe005 // alt-t transpose current word with previous

    KeyDeleteRightWord = 0xe006 // alt-d delete to end of word start at cursor
    KeyDeleteLeftChar  = '\b'   // ctrl-h cut previous character
    KeyDeleteLeftWord  = '\x17' // ctrl-w cut previous word, also ctrl-bs on linux and ctrl-del on bsd
    KeyDeleteLineStart = '\x15' // ctrl-u cut from beginning of line to cursor
    KeyDeleteLineEnd   = '\v'   // ctrl-k cut from cursor to end of line

    KeyUndo  = '\x1f' // ctrl-? undo last edit
    KeyPaste = '\x19' // ctrl-y paste

    KeySuspend = '\x1a' // ctrl-z suspend process
```

I was intending to start with the missing `Move` and `Delete` sequences but have not had time.

**Comment by ahmetb** [comment]
I realized bash and zsh work differently for some of these shortcuts. For example `KeyDeleteLineStart` (ctrl+U) deletes the whole line in ZSH, but does what's described above for Bash.

This library probably should implement a minimal set that just does the main tasks and not more. If you are good at Node.js, I recommend looking at Inquirer.js if they've got any magic to handle this aspect.

**Comment by CyrusNajmabadi** [comment]
Hi @AlecAivazis. We've also run into this over in Pulumi with https://github.com/pulumi/pulumi/issues/1565.   It would definitely be great if these could ever be handled.  Thanks!
